frontend/packages/core/src/AppLayout/shortLinker.tsx

```diff
@@ -0,0 +1,193 @@
+      background: "#2d3db4",
+    },
+    "&:active": {
+      background: "#2938a5",
+    },
+  },
+  props => ({
+    background: props.$open ? "#2d3db4" : "unset",
+  })
+);
+
+/**
+ * Component that will display a Button to generate a short link
+ * On click the component will do the following:
+ * - Read in the data from the temporary storage in the WorkflowStorageContext
+ * - Rotate it into a readable format for the API
+ * - Send an API request asking for it to be stored
+ * - If successful,
+ * -     Will switch to a readable only input field with a clipboard button to copy the short link
+ *       (this is the only time it is ever displayed)
+ * - If not successful,
+ * -     Will display a toast error message for the user
+ */
+const ShortLinker = () => {
+  const { workflows } = useAppContext();
+  const { removeWorkflowSession, retrieveWorkflowSession } = useShortLinkContext();
+  const [open, setOpen] = React.useState(false);
+  const [shortLink, setShortLink] = React.useState<string | null>(null);
+  const [validWorkflow, setValidWorkflow] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<ClutchError | null>(null);
+  const anchorRef = React.useRef(null);
+  const location = useLocation();
+
+  const checkValidWorkflow = () => {
+    const workflow = workflowByRoute(workflows, location.pathname);
+
+    setValidWorkflow(workflow?.shortLink ?? false);
+  };
+
+  // will trigger on a location change, emptying out our temporary storage and rechecking the workflow
+  React.useEffect(() => {
+    if (workflows.length) {
+      checkValidWorkflow();
+    }
+
+    removeWorkflowSession();
+  }, [location]);
+
+  // Used for initial load to verify that once our workflows have loaded we are on a valid workflow
+  React.useEffect(() => {
+    if (workflows.length) {
+      checkValidWorkflow();
+    }
+  }, [workflows]);
+
+  const handleToggle = () => {
+    setOpen(!open);
+    setShortLink(null);
+  };
+
+  const handleClose = event => {
+    if (anchorRef.current && anchorRef.current.contains(event.target)) {
+      return;
+    }
+    setOpen(false);
+  };
+
+  // Will rotate our object into an array of type IShareableState to send in the API request
+  const rotateStore = (sessionStore: HydratedData): IClutch.shortlink.v1.IShareableState[] =>
+    Object.keys(sessionStore).map(key => ({ key, state: sessionStore[key] }));
+
+  const generateShortLink = () => {
+    const requestData: IClutch.shortlink.v1.ICreateRequest = {
+      path: `${location.pathname}${location.search}`,
+      state: rotateStore(retrieveWorkflowSession()),
+    };
+
+    client
+      .post("/v1/shortlink/create", requestData)
+      .then(response => {
+        const { hash } = response.data as IClutch.shortlink.v1.ICreateResponse;
+        setShortLink(`${window.location.origin}/sl/${hash}`);
+      })
+      .catch((err: ClutchError) => {
+        console.warn("failed to generate short link", err); // eslint-disable-line
```

Comment:
Do we need this console warn if we are showing an error to the users?

Reply:
This was because the toast doesn't contain a lot of information (below), so it was more for debugging purposes.